webrtc.html

```diff
@@ -5624,10 +5641,9 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
             </p>
           </li>
           <li>
-            <p>Queue a task to fire an event named <code title=
-            "event-track"><a>track</a></code> with <var>transceiver</var>,
-            <var>track</var>, and <var>streams</var> at the
-            <var>connection</var> object.</p>
+            <p>Add a new <code><a>RTCTrackEvent</a></code> with
+            <var>transceiver</var>, <var>track</var>, and
+            <var>streams</var> to <var>trackEvents</var>.</p>
           </li>
         </ol>
         <p>To <dfn id="process-remote-track-removal">
```

Comment:
I think it's generally the same as most other software development. If you have a use case for the non-queueing versions, then yeah, keep that outside. Otherwise, to deduplicate, feel free to move it inside the algorithm.

To be explicit, if you have

```js
function caller1() {
  // ...
  wrapperOperation(doStuff);
  // ...
}

function caller2() {
  // ...
  wrapperOperation(doStuff);
  // ...
}
```

then maybe you should refactor to a single function `doWrapperOperationAndDueStuff()` and call that instead. But if you have a `caller3` that directly calls `doStuff()`, then don't refactor.